src/main/java/cz/cvut/kbss/termit/persistence/dao/TermDao.java

```diff
@@ -376,6 +379,42 @@ private List<TermDto> executeAndBuildHierarchy(TypedQuery<FlatTermDto> query) {
         return result;
     }
 
+    /**
+     * Finds all terms in the specified vocabulary, regardless of their position in the term hierarchy and returns
+     * them as a flat list of DTOs.
+     *
+     * @param vocabulary Vocabulary whose terms to retrieve. A reference is sufficient
+     * @param pageSpec   Page specification
+     * @return Flat list of vocabulary term DTOs
+     */
+    public List<FlatTermDto> findAllFlat(Vocabulary vocabulary, Pageable pageSpec) {
+        Objects.requireNonNull(vocabulary);
+        try {
+            final TypedQuery<FlatTermDto> query =
+                    em.createNativeQuery("SELECT DISTINCT ?term WHERE {" +
+                                      "GRAPH ?context { " +
+                                      "?term a ?type ;" +
+                                      "?hasLabel ?label ;" +
+                                      "FILTER (lang(?label) = ?labelLang) ." +
+                                      "}" +
+                                      "?term ?inVocabulary ?vocabulary ." +
+                                      " } ORDER BY " + orderSentence("?label"),
+                              FlatTermDto.class)
+                      .setParameter("context", context(vocabulary))
+                      .setParameter("type", typeUri)
+                      .setParameter("vocabulary", vocabulary.getUri())
+                      .setParameter("hasLabel", LABEL_PROP)
+                      .setParameter("inVocabulary", TERM_FROM_VOCABULARY)
+                      .setParameter("labelLang", config.getLanguage())
```

Comment:
Could you adjust the queries so that they use the vocabulary-based language rather than the config-based one (recent merge in main, see the non-flat version of this method above)?